src/Extension/CoreExtension.php

```diff
@@ -1298,22 +1298,18 @@ function twig_length_filter(Environment $env, $thing)
         }
 
         if (is_scalar($thing)) {
-            return mb_strlen($thing, $env->getCharset());
+            return mb_strlen((string) $thing, $env->getCharset());
         }
 
-        if ($thing instanceof \SimpleXMLElement) {
+        if ($thing instanceof \Countable || \is_array($thing) || $thing instanceof \SimpleXMLElement) {
             return \count($thing);
         }
 
-        if (\is_object($thing) && method_exists($thing, '__toString') && !$thing instanceof \Countable) {
+        if (\is_object($thing) && method_exists($thing, '__toString')) {
             return mb_strlen((string) $thing, $env->getCharset());
         }
 
-        if ($thing instanceof \Countable || \is_array($thing)) {
-            return \count($thing);
-        }
-
-        if ($thing instanceof \IteratorAggregate) {
+        if ($thing instanceof \Traversable) {
```

Comment:
I would do that before using `__toString` (also to be BC).

Reply:
good catch, done and I've added update the test to catch this in the future : )